common/lib/xmodule/xmodule/tabs.py

```diff
@@ -168,610 +161,30 @@ def from_json(tab_dict):
-        super(TextbookTabsBase, self).__init__(
-            name=_("Textbooks"),
-            tab_id=tab_id,
-            link_func=None,
-        )
-
-    @abstractmethod
-    def items(self, course):
-        """
-        A generator for iterating through all the SingleTextbookTab book objects associated with this
-        collection of textbooks.
-        """
-        pass
-
-
-class TextbookTabs(TextbookTabsBase):
-    """
-    A tab representing the collection of all textbook tabs.
-    """
-    type = 'textbooks'
-
-    def __init__(self, tab_dict=None):  # pylint: disable=unused-argument
-        super(TextbookTabs, self).__init__(
-            tab_id=self.type,
-        )
-
-    def can_display(self, course, settings, is_user_authenticated, is_user_staff, is_user_enrolled):
-        return settings.FEATURES.get('ENABLE_TEXTBOOK')
-
-    def items(self, course):
-        for index, textbook in enumerate(course.textbooks):
-            yield SingleTextbookTab(
-                name=textbook.title,
-                tab_id='textbook/{0}'.format(index),
-                link_func=lambda course, reverse_func, index=index: reverse_func(
-                    'book', args=[course.id.to_deprecated_string(), index]
-                ),
-            )
-
-
-class PDFTextbookTabs(TextbookTabsBase):
-    """
-    A tab representing the collection of all PDF textbook tabs.
-    """
-    type = 'pdf_textbooks'
-
-    def __init__(self, tab_dict=None):  # pylint: disable=unused-argument
-        super(PDFTextbookTabs, self).__init__(
-            tab_id=self.type,
-        )
-
-    def items(self, course):
-        for index, textbook in enumerate(course.pdf_textbooks):
-            yield SingleTextbookTab(
-                name=textbook['tab_title'],
-                tab_id='pdftextbook/{0}'.format(index),
-                link_func=lambda course, reverse_func, index=index: reverse_func(
-                    'pdf_book', args=[course.id.to_deprecated_string(), index]
-                ),
-            )
-
-
-class HtmlTextbookTabs(TextbookTabsBase):
-    """
-    A tab representing the collection of all Html textbook tabs.
-    """
-    type = 'html_textbooks'
-
-    def __init__(self, tab_dict=None):  # pylint: disable=unused-argument
-        super(HtmlTextbookTabs, self).__init__(
-            tab_id=self.type,
-        )
-
-    def items(self, course):
-        for index, textbook in enumerate(course.html_textbooks):
-            yield SingleTextbookTab(
-                name=textbook['tab_title'],
-                tab_id='htmltextbook/{0}'.format(index),
-                link_func=lambda course, reverse_func, index=index: reverse_func(
-                    'html_book', args=[course.id.to_deprecated_string(), index]
-                ),
+        # TODO: don't import openedx capabilities from common
+        from openedx.core.djangoapps.course_views.course_views import CourseViewTypeManager
+        tab_type_name = tab_dict.get('type')
```

Comment:
If there's no tab type, is that immediately an error? (Perhaps one that could have a more explicit error message).